docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+  This person cares about all hardware and its associated hardware and
+  cluster-wide software resources.
+
+- **Backend Service Owner**: The person responsible for managing the Backend Service
+  on top of the cluster provided by the Cluster Administrator.
+
+- **InstructLab Model Developer**: This person is a user of the Backend Service.
+  They don't care about the underlying infrastructure details. They care about
+  data sets, model training, model evaluation, and all of those types of related
+  details associated with the pipeline implemented by the Backend Service.
+  - **Sub-category A**: A Model Developer who is trying to add domain-specific
+  knowledge to the model and is very interested in how they have improved (or
+  not) the model in that specific domain
+  - **Sub-category B**: A Model Developer who is interested in improving an
+  overall model benchmark
+
+## Guiding Principles for the Future
+
+We aim to articulate a set of guiding principles that will help guide our
+technical decision-making as we evolve the Backend Service. These principles are
+intended to be high-level and should be used to inform the design of the system
+as a whole.
+
+When faced with a difficult technical decision with a set of trade-offs, we should
+review these principles to ensure we make a decision that aligns with them.
+We should also consider these principles to be open to change as we learn more,
+but they will always serve as a guide.
+
+Summarized, these principles are:
+
+- Scalable
+- Service-Oriented
+- Observable
+- Reusable
+- Small Iterations
+- Minimize Mandatory External Dependencies
+- Make Development Accessible
+- Model Independence
+
+### Scalable
+
+The initial iteration of the Backend Service targeted a particular set of clusters
+with a known set of hardware resources. As we evolve to become a more flexible
+and reusable system, we must design things in a way that allows us to scale up
+or down as needed.
+
+We need to drive down the minimum requirements while also ensuring we can scale
+up to much larger environments with more resources available and make full use
+of those resources.
+
+As a minimum target, we should be able to run some configuration of the full
+backend pipeline on a single machine. We can use these as minimum requirements:
+
+- Linux server with 4x H100 GPUs
+
+### Service-Oriented
+
+While considering the future, we must understand each targeted persona and
+design services that are focused on their needs.
+
+We should consider the persona of an InstructLab Model Developer and what
+actions and information they care about. That should drive the service
+interface provided by our Backend Service. Some examples of this point:
+
+- An InstructLab Model Developer cares about defining inputs to our pipeline,
+  launching that pipeline, following its status, and consuming its outputs.
+
+- An InstructLab Model Developer does not care about the exact details of how
```

Comment:
In RHEL AI systems there will be no Kubernetes, as of now.

Reply:
This is an upstream doc. RHEL AI is out of scope.

The doc is also focused on what we need to build to have an open and reproducible full backend that can run on a full cluster. The tools used to produce the full model builds out of the project are not open and not reusable in their current form anyway. Evolving and replacing that is the primary goal here. It just so happens that we could scale down the same tools to a single server, as well.